### PR TITLE
Add Obj format in MeshReader including colors

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description
Add Obj format in MeshReader including colors. 
All DgtalTools viewer will be able to read and display obj mesh without any changes.
It is based on the same code of SurfaceMeshReader with adaptation by removing normal read and adding mtl file reading to recover colors.
+ It fix an error on obj file with relative face reference (also applied in SurfaceMeshReader)

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by kerautret** [comment]
Codacy issue looks a bug or I miss something...
The two spaces are well here:
Capture d’écran 2021-07-15 à 22 32 05

**Comment by dcoeurjo** [comment]
travis is back...

File: src/DGtal/io/readers/MeshReader.h [hunk 1]
```diff
@@ -127,10 +127,36 @@ struct MeshReader
...
+  bool importOBJFile(const std::string & filename, DGtal::Mesh<TPoint> & aMesh );
 
+  
+  /// Checks that every index in \a indices are different from the others.
+  /// @param indices a vector of integer indices
+  /// @return 'true' iff the integer indices are all pairwise different.
+  static
+  bool verifyIndicesUniqueness( const std::vector< unsigned int > &indices );
+
+
```
Comment (dcoeurjo) [inline src/DGtal/io/readers/MeshReader.h:148]:
```suggestion
```

**Review by dcoeurjo — COMMENTED** [review]
Thx for the PR. Just some comments

**Comment by dcoeurjo** [comment]
Please use the clang format to have the proper code formatting (there is a .clang-format in the project)

**Comment by dcoeurjo** [comment]
Please use the clang format to have the proper code formatting (there is a .clang-format in the project)

File: src/DGtal/io/readers/MeshReader.h [hunk 2]
```diff
@@ -127,10 +127,35 @@ struct MeshReader
...
+  /// Reads an input file as an OBJ file format and outputs the
+  /// corresponding surface mesh.
+  ///
+  /// @param[in] filename the input filename of the OBJ file to be read.
+  /// @param[out] aMesh the output  mesh.
+  ///
+  /// @return 'true' if both reading the input stream was ok and the
+  /// created mesh is ok.
+  static
+  bool importOBJFile(const std::string & filename, DGtal::Mesh<TPoint> & aMesh );
```
[the remark this answers is not in the document]
Comment (kerautret) [reply src/DGtal/io/readers/MeshReader.h:139]: no it is static ;)

[the remark this answers is not in the document]
Comment (dcoeurjo) [reply src/DGtal/io/readers/MeshReader.h:139]: 👍

File: src/DGtal/io/readers/MeshReader.ih [hunk 3]
```diff
@@ -274,16 +275,129 @@ DGtal::MeshReader<TPoint>::importOFSFile(const std::string & aFilename,
...
+  for ( ; input.good() && ! input.eof(); std::getline( input, linestr ), l++ )
+    {
+      if ( linestr.empty() ) continue; // skip empty line
+      if ( linestr[0] == '#' ) continue; // skip comment line
+      std::istringstream lineinput( linestr );
+      std::operator>>( lineinput, keyword ); // lineinput >> keyword;
+      if ( keyword == "v" ) {
+        lineinput >> p[ 0 ] >> p[ 1 ] >> p[ 2 ];
+        vertices.push_back( p );
+      } else if ( keyword == "f" ) {
```
[the remark this answers is not in the document]
Comment (kerautret) [reply src/DGtal/io/readers/MeshReader.ih]: done ;)

File: src/DGtal/io/readers/MeshReader.ih [hunk 4]
```diff
@@ -292,7 +406,52 @@ DGtal::MeshReader<TPoint>::importOFSFile(const std::string & aFilename,
 
 
 
+//-----------------------------------------------------------------------------
+template <typename TPoint>
+bool
+DGtal::MeshReader<TPoint>::
+verifyIndicesUniqueness( const std::vector< unsigned int > &indices )
+{
+  std::set<unsigned int> sindices( indices.begin(), indices.end() );
```
[the remark this answers is not in the document]
Comment (kerautret) [reply src/DGtal/io/readers/MeshReader.ih]: done thanks

**Comment by dcoeurjo** [comment]
Ok the Exclude is not working...

**Review by dcoeurjo — APPROVED** [review]

**Pull request merged**